[DJM] Add callout to DJM in airflow docs (#19590)
DataDog/integrations-core: airflow/README.md

# Agent Check: Airflow

<div class="alert alert-info">

> <a href="https://docs.datadoghq.com/data_jobs/">Data Jobs Monitoring</a> helps you observe, troubleshoot  and understand lineage in Airflow DAGs and tasks.<br/><br/>

Reviewer: ```suggestion
<a href="https://docs.datadoghq.com/data_jobs/">Data Jobs Monitoring (in preview for Airflow)</a> provides tracing for Airflow DAG runs out-of-the-box so you can quickly troubleshoot problematic tasks, correlate DAG runs to logs, and understand complex pipelines with data lineage across DAGs.<br/><br/>
```

> This page is limited to documentation for ingesting Airflow data using the API integrations and the data dog agent.

Reviewer: ```suggestion
This page is limited to documentation for collecting Airflow integration metrics & logs with the Datadog Agent.
```

> <a href="https://docs.datadoghq.com/data_jobs/">Data Jobs Monitoring (in preview for Airflow)</a> provides tracing for Airflow DAG runs out-of-the-box so you can quickly troubleshoot problematic tasks, correlate DAG runs to logs, and understand complex pipelines with data lineage across DAGs.<br/><br/>
> This page is limited to documentation for collecting Airflow integration metrics & logs with the Datadog Agent.

Reviewer 2: ```suggestion
<a href="https://docs.datadoghq.com/data_jobs/">Data Jobs Monitoring</a> (in Preview for Airflow) provides out-of-the-box tracing for Airflow DAG runs, helping you quickly troubleshoot problematic tasks, correlate DAG runs to logs, and understand complex pipelines with data lineage across DAGs.<br/><br/>
<strong>Note</strong>: This page covers only the documentation for collecting Airflow integration metrics and logs using the Datadog Agent.
```
